packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -104,6 +138,71 @@ const modifiers = [
   },
 ];
 
+const ListButton = ({ icon, format, label }) => {
+  const editor = useSlate();
+
+  /**
+   *
+   * @param {import('slate').Node} node
+   * @returns boolean
+   */
+  const isListNode = (node) => {
+    return !Editor.isEditor(node) && Element.isElement(node) && node.type === 'list';
+  };
+
+  const isListActive = () => {
+    const { selection } = editor;
+
+    if (!selection) return false;
+
+    const [match] = Array.from(
+      Editor.nodes(editor, {
+        at: Editor.unhangRange(editor, selection),
+        match: (node) => isListNode(node) && node.format === format,
+      })
+    );
+
+    return Boolean(match);
+  };
+
+  const isActive = isListActive();
+
+  const toggleList = () => {
+    Transforms.unwrapNodes(editor, {
+      match: (node) => isListNode(node) && ['ordered', 'unordered'].includes(node.format),
+      split: true,
+    });
+
+    Transforms.setNodes(editor, {
+      type: isActive ? 'paragraph' : 'list-item',
+    });
+
+    if (!isActive) {
+      const block = { type: 'list', format, children: [] };
```

Comment:
I saw some weird behaviour when I toggle list, 
number 2 is overlapping with another(could be a slate bug).

https://github.com/strapi/strapi/assets/7886244/b6b17367-1546-4b0a-a2e9-30707f83705c

Reply:
Also if we need to add nested item, it doesnt work , with schema it works.

Reply:
Yes that's very weird. But honestly I would rather just write that down in a list of bugs and move forward, beause I think once we'll have the basic functionality in the feature branch it will be easier to do the QA and all the fixes. If we try to address all bugs now I'm afraid it will be hard to move forward and to be ready for the alpha

Also I suspect this could be just a rendering bug, we'd have to check if the slate state to know.

Reply:
sounds good to me